core/src/main/java/org/elasticsearch/common/network/NetworkService.java

```diff
@@ -273,4 +270,19 @@ public InetAddress resolvePublishHostAddresses(String publishHosts[]) throws IOE
         }
         return InetAddress.getAllByName(host);
     }
+
+    /**
+     * Register custom name resolver a DiscoveryPlugin might provide
+     * @param discoveryPlugins Discovery plugins
+     */
+    public static List<CustomNameResolver> registerCustomNameResolvers(Settings settings, List<DiscoveryPlugin> discoveryPlugins) {
```

Comment:
@rjernst Totally agreed. Was a left over.
About making it private, I can't because its also used by TransportClient.
May be I should move it in a NodeUtils class or something? Or better in `DiscoveryPlugin` interface? WDYT?